api/class-wc-calypso-bridge-product-reviews-controller.php

```diff
@@ -0,0 +1,385 @@
+			'no_found_rows' => false,
+		);
+
+		if ( ! empty( $request['product']) ) {
+			$args['post_id'] = $request['product'];
+		}
+
+		if ( ! empty( $request['search']) ) {
+			$args['search'] = $request['search'];
+		}
+
+		$reviews_query = new WP_Comment_Query;
+		$reviews       = $reviews_query->query( $args );
+
+		$data = array();
+		foreach ( $reviews as $review_data ) {
+			$review = $this->prepare_item_for_response( $review_data, $request );
+			$review = $this->prepare_response_for_collection( $review );
+			$data[] = $review;
+		}
+
+		$response = rest_ensure_response( $data );
+
+		$response->header( 'X-WP-Total', (int) $reviews_query->found_comments );
+		$response->header( 'X-WP-TotalPages', (int) $reviews_query->max_num_pages );
+
+		return $response;
+	}
+
+	/**
+	 * Prepare links for the request.
+	 *
+	 * @param WP_Comment $review Product review object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return array Links for the given product review.
+	 */
+	protected function prepare_links( $review, $request ) {
+		$links = array(
+			'self' => array(
+				'href' => rest_url( sprintf( '/%s/%s', $this->namespace, 'products/' . $review->comment_post_ID . '/reviews/' . $review->comment_ID ) ),
+			),
+			'collection' => array(
+				'href' => rest_url( sprintf( '/%s/%s', $this->namespace, 'products/' . $review->comment_post_ID . '/reviews' ) ),
+			),
+			'up' => array(
+				'href' => rest_url( sprintf( '/%s/%s', $this->namespace, 'products/' . $review->comment_post_ID ) ),
+			),
+		);
+
+		return $links;
+	}
+
+	/**
+	 * Prepare a single product review output for response.
+	 *
+	 * v3 adds 'product_id' and 'approved'.
+	 *
+	 * @param WP_Comment $review Product review object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response $response Response data.
+	 */
+	public function prepare_item_for_response( $review, $request ) {
+		$product = new WC_Product( $review->comment_post_ID );
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+
+		$images        = wp_get_attachment_image_src( $product->get_image_id(), 'full' );
+		$product_image = is_array( $images ) ? current( $images ) : '';
+
+		switch( $review->comment_approved ) {
+			case '1':
+				$status = 'approved';
+				break;
+			case 'spam':
+				$status = 'spam';
+				break;
+			case 'trash':
+				$status = 'trash';
+				break;
+			case '0':
+			default:
+				$status = 'pending';
+				break;
+		}
+
+		$content = 'view' === $context ? wpautop( $review->comment_content ) : $review->comment_content;
+
+		$data = array(
+			'id'               => (int) $review->comment_ID,
+			'date_created'     => wc_rest_prepare_date_response( $review->comment_date ),
+			'date_created_gmt' => wc_rest_prepare_date_response( $review->comment_date_gmt ),
+			'review'           => $content,
+			'rating'           => (int) get_comment_meta( $review->comment_ID, 'rating', true ),
+			'name'             => $review->comment_author,
```

Comment:
[Admins that can manage WooCommerce](https://github.com/Automattic/wc-calypso-bridge/pull/50/files#diff-1df620e4fab840d4ccd911cff40ad4adR58).

This endpoint is used for the product reviews management screen in Calypso, so we show the comment authors name. Email is used for avatar.

These exact fields are already exposed over the standard WooCommerce endpoints `/products/$product_id/reviews` for the same permissions.

Reply:
Just looked and you're right, it is `avatar_urls`. 

Are you OK with me leaving it as is? This is a direct carry over of the API that was in `wc-api-dev`, and we are also exposing the email over `/v2`, and `/v1` versions of `/products/$id/reviews`  as well. We aren't exposing these fields to just anyone -- you need API keys. 

Proton is working on the mergeable version of a global products reviews endpoint like this one -- this PR is a stepping stone until we can migrate Calypso to the final API. If we want to discuss removing email and name fields there we can -- but I feel like that makes the API less useful (and again these fields are exposed to the same people in wp-admin).

Reply:
I think it is fine to keep this as-is since it is what is currently in place, and as you point out will be eventually deprecated in favor of a core endpoint.